interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -215,46 +201,85 @@ private void run() {
-      tester.largeUnary();
-    } else if ("client_streaming".equals(testCase)) {
-      tester.clientStreaming();
-    } else if ("server_streaming".equals(testCase)) {
-      tester.serverStreaming();
-    } else if ("ping_pong".equals(testCase)) {
-      tester.pingPong();
-    } else if ("empty_stream".equals(testCase)) {
-      tester.emptyStream();
-    } else if ("compute_engine_creds".equals(testCase)) {
-      tester.computeEngineCreds(defaultServiceAccount, oauthScope);
-    } else if ("service_account_creds".equals(testCase)) {
-      String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
-      FileInputStream credentialsStream = new FileInputStream(new File(serviceAccountKeyFile));
-      tester.serviceAccountCreds(jsonKey, credentialsStream, oauthScope);
-    } else if ("jwt_token_creds".equals(testCase)) {
-      FileInputStream credentialsStream = new FileInputStream(new File(serviceAccountKeyFile));
-      tester.jwtTokenCreds(credentialsStream);
-    } else if ("oauth2_auth_token".equals(testCase)) {
-      String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
-      FileInputStream credentialsStream = new FileInputStream(new File(serviceAccountKeyFile));
-      tester.oauth2AuthToken(jsonKey, credentialsStream, oauthScope);
-    } else if ("per_rpc_creds".equals(testCase)) {
-      String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
-      FileInputStream credentialsStream = new FileInputStream(new File(serviceAccountKeyFile));
-      tester.perRpcCreds(jsonKey, credentialsStream, oauthScope);
-    } else if ("unimplemented_method".equals(testCase)) {
-      tester.unimplementedMethod();
-    } else if ("cancel_after_begin".equals(testCase)) {
-      tester.cancelAfterBegin();
-    } else if ("cancel_after_first_response".equals(testCase)) {
-      tester.cancelAfterFirstResponse();
-    } else  if ("timeout_on_sleeping_server".equals(testCase)) {
-      tester.timeoutOnSleepingServer();
-    } else {
-      throw new IllegalArgumentException("Unknown test case: " + testCase);
+  private void runTest(TestCases testCase) throws Exception {
+    switch (testCase) {
+      case EMPTY_UNARY:
+        tester.emptyUnary();
+        break;
+
+      case LARGE_UNARY:
+        tester.largeUnary();
+        break;
+
+      case CLIENT_STREAMING:
+        tester.clientStreaming();
+        break;
+
+      case SERVER_STREAMING:
+        tester.serverStreaming();
+        break;
+
+      case PING_PONG:
+        tester.pingPong();
+        break;
+
+      case EMPTY_STREAM:
+        tester.emptyStream();
+        break;
+
+      case COMPUTE_ENGINE_CREDS:
+        tester.computeEngineCreds(defaultServiceAccount, oauthScope);
+        break;
+
+      case SERVICE_ACCOUNT_CREDS: {
+        String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
```

Comment:
It is in later versions of java with StandardCharset.  Seeing string literals makes me afraid, so I avoid them whenever possible.